### DISCUSSION TITLE
Configure LDAP (Logging)

### DISCUSSION THREAD

**Discussion by Limerick-gh** [modoboa/modoboa#2776] [Q&A]
Hi there
I am looking for help regarding LDAP configuration.

I used the installer on Ubuntu  22.04 and went through the steps in documentation to setup LDAP authentication and synchronization to connect to internal test AD/LDAP server.
Unfortunately, when trying to login, reply is "username and password didn't match".
I also set logging according to https://django-auth-ldap.readthedocs.io/en/latest/logging.html#logging in order to get some more info. But /var/log/auth.log doesn't show anything helpful.

1. Am I looking at the right log file? If not, where to look at?
2. Has anybody a working setup with AD/LDAP and can share the settings of LDAP parameters?

Any hints would be much appreciated!

**Comment by tonioo** [comment]
About logging, it will actually depend on your global configuration... Have you enabled syslog handler?
And about LDAP settings, what have you tried?

**Reply by Limerick-gh** [reply]
Thanks for your reply. 
Thought it would log out of the box by simply adding the lines mentioned, will double check regarding syslog.
Regarding LDAP, I tried with hostname, IP address as well as different filters. But I guess I have to sort the logging first. Will reply once I get it.

**Comment by Aegis8** [comment]
@Limerick-gh 

Check the following section of the /etc/dovecot/dovecot-ldap.conf.ext file:

#Filter for user lookup. Some variables can be used (see

#http://wiki2.dovecot.org/Variables for full list):

#%u - username

#%n - user part in user@domain, same as %u if there's no domain

#%d - domain part in user@domain, empty if user there's no domain


I had the miss match due to which uid is used for the lookups and to bind to LDAP, by default dovecot/Modoboa will use the e-mail for the uid.

As an example here are my relevant settings:
auth_bind = yes
auth_bind_userdn = uid=%n,dc=example,dc=org
user_filter = (&(objectClass=asrequired)(uid=%n)(mail=%u))
pass_filter = (&(objectClass=asrequired)(uid=%n)(mail=%u))

Our LDAP integration is not active now, so I might have missed something. Hope this helps though.
